OMEdit/OMEditLIB/OMC/OMCProxy.cpp

```diff
@@ -573,9 +574,14 @@ bool OMCProxy::printMessagesStringInternal()
   /* Loop in reverse order since getMessagesStringInternal returns error messages in reverse order. */
   for (int i = errorsSize; i > 0 ; i--) {
     setCurrentError(i);
-    MessageItem messageItem(MessageItem::Modelica, getErrorFileName(), getErrorReadOnly(), getErrorLineStart(), getErrorColumnStart(), getErrorLineEnd(),
-                            getErrorColumnEnd(), getErrorMessage(), getErrorKind(), getErrorLevel());
-    MessagesWidget::instance()->addGUIMessage(messageItem);
+    const int errorId = getErrorId();
+    if (errorId == 371 || errorId == 372 || errorId == 373) {
```

Comment:
@perost I am using the error ids to intercept the error and show dialog to the user. Is this the correct way? I believe better than string matching.

Reply:
That's probably the best you can do as it is now. There's nothing stopping the ids from changing, and sometimes people change old ids to e.g. close gaps in the numbering when removing deprecated flags, but string matching has the same problem.